test/extended/cluster/metrics.go

```diff
@@ -0,0 +1,56 @@
+package cluster
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"time"
+)
+
+const (
+	marker_name string = "cluster_loader_marker"
+)
+
+type Metrics interface {
+	printLog() error
+}
+
+type BaseMetrics struct {
+	// To let the 3rd party know that this log entry is important
+	// TODO set this up by config file
+	Marker string `json:"marker"`
+	Name   string `json:"name"`
+	Type   string `json:"type"`
+}
+
+type TestDuration struct {
+	BaseMetrics
+	StartTime    time.Time `json:"startTime"`
+	TestDuration string    `json:"testDuration"`
+}
+
+func (td TestDuration) printLog() error {
+	b, err := json.Marshal(td)
+	fmt.Println(string(b))
+	return err
+}
+
+func LogMetrics(metrics []Metrics) error {
+	for _, m := range metrics {
+		err := m.printLog()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func NewTestDuration(name string, startTime time.Time, testDuration time.Duration) TestDuration {
+	m := TestDuration{}
+	m.Marker = marker_name
+	m.Name = name
+	m.Type = reflect.TypeOf(m).Name()
+	m.StartTime = startTime
+	m.TestDuration = fmt.Sprintf("%s", testDuration.String())
+	return m
```

Comment:
Just a preference for this, return a new `TestDuration` struct in line.

Reply:
Done.